### PR TITLE
Reporter can add a facility

### PR DESCRIPTION
**Pull request by dleard**
- Created an 'AddFacility' component
 - added a createFacility mutation

### CODE REVIEW

File: app/components/facility/AddFacilityComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
+import React, {useState} from 'react';
+import {Button, Modal} from 'react-bootstrap';
+import {JSONSchema6} from 'json-schema';
+import JsonSchemaForm, {IChangeEvent} from 'react-jsonschema-form';
+import moment from 'moment-timezone';
+import FormFieldTemplate from '../../containers/Forms/FormFieldTemplate';
```
Comment (matthieu-foucault) [inline app/components/facility/AddFacilityComponent.tsx]:
```suggestion
import FormFieldTemplate from 'containers/Forms/FormFieldTemplate';
```

File: app/components/facility/AddFacilityComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+import FormObjectFieldTemplate from '../../containers/Forms/FormObjectFieldTemplate';
```
Comment (matthieu-foucault) [inline app/components/facility/AddFacilityComponent.tsx]:
```suggestion
import FormObjectFieldTemplate from 'containers/Forms/FormObjectFieldTemplate';
```

File: app/components/facility/AddFacilityComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+interface Props {
+  handleAddFacility: any;
+  organisationRowId: string;
+}
+
+export const AddFacility = props => {
```
Comment (matthieu-foucault) [inline app/components/facility/AddFacilityComponent.tsx]: Rename the file to `AddFacility` and make this the default export

File: app/components/facility/AddFacilityComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+import {Button, Modal} from 'react-bootstrap';
+import {JSONSchema6} from 'json-schema';
+import JsonSchemaForm, {IChangeEvent} from 'react-jsonschema-form';
+import moment from 'moment-timezone';
+import FormFieldTemplate from '../../containers/Forms/FormFieldTemplate';
+import FormObjectFieldTemplate from '../../containers/Forms/FormObjectFieldTemplate';
+
+interface Props {
+  handleAddFacility: any;
+  organisationRowId: string;
```
Comment (matthieu-foucault) [inline app/components/facility/AddFacilityComponent.tsx]: Shouldn't this be a number?

File: app/components/facility/AddFacilityComponent.tsx [hunk 5]
```diff
@@ -0,0 +1,134 @@
+import React, {useState} from 'react';
+import {Button, Modal} from 'react-bootstrap';
+import {JSONSchema6} from 'json-schema';
+import JsonSchemaForm, {IChangeEvent} from 'react-jsonschema-form';
+import moment from 'moment-timezone';
+import FormFieldTemplate from '../../containers/Forms/FormFieldTemplate';
+import FormObjectFieldTemplate from '../../containers/Forms/FormObjectFieldTemplate';
+
+interface Props {
+  handleAddFacility: any;
```
Comment (matthieu-foucault) [inline app/components/facility/AddFacilityComponent.tsx]: The prop name should be `onAddFacility` The `handleX` convention is for methods inside of the component

File: app/components/facility/AddFacilityComponent.tsx [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+        }
+      }
+    }
+  };
+
+  const customFormats = {
+    'postal-code': /[A-Z]\d[A-Z]\s?\d[A-Z]\d/i
+  };
+
+  const reportingYear = moment().format('YYYY');
```
Comment (matthieu-foucault) [inline app/components/facility/AddFacilityComponent.tsx]: This value should come from the `getReportingYear` query

File: app/containers/Facilities/FacilitiesListContainer.tsx [hunk 7]
```diff
@@ -1,6 +1,8 @@
-import React, {useEffect} from 'react';
+import React, {useEffect, useState} from 'react';
 import {graphql, createRefetchContainer} from 'react-relay';
 import SearchTableLayout from 'components/SearchTableLayout';
+import createFacilityMutation from 'mutations/facility/createFacilityMutation';
+import {AddFacility} from '../../components/facility/AddFacilityComponent';
 import FacilitiesRowItemContainer from './FacilitiesRowItemContainer';
 
 export const FacilitiesList = props => {
```
Comment (matthieu-foucault) [inline app/containers/Facilities/FacilitiesListContainer.tsx]: Add some types here please

File: app/components/facility/AddFacilityComponent.tsx (hunk 4 above)
Comment (dleard) [reply app/components/facility/AddFacilityComponent.tsx]: yep

Comment (dleard) [reply app/components/facility/AddFacilityComponent.tsx]: Hm. On second look, our facility table stores the reportingYear as a string. It's also stored as a varchar in swrs and used as a varchar in the init functions

File: app/containers/Facilities/FacilitiesListContainer.tsx (hunk 7 above)
Comment (dleard) [reply app/containers/Facilities/FacilitiesListContainer.tsx]: I added some types, did I miss any?

File: app/components/facility/AddFacilityComponent.tsx (hunk 4 above)
Comment (matthieu-foucault) [reply app/components/facility/AddFacilityComponent.tsx]: uh? This is 'organisationRowId', not reporting year. Also reportingYear should be a number everywhere, including in SWRS, where it was changed to a varchar only to accomodate a metabase bug

Comment (dleard) [reply app/components/facility/AddFacilityComponent.tsx]: oh yeah! Well I changed the organisationRowId to a number, but we do have a type problem with reportingYear, it is a varchar in most places, and an int in others

Comment (dleard) [reply app/components/facility/AddFacilityComponent.tsx]: Note added for checking into reportingYear type mismatch: https://youtrack.button.is/agiles/101-1/102-295?issue=GGIRCS-1111

**Pull request merged**